engine/src/dot/http_rpc.rs

```diff
@@ -151,13 +151,21 @@ impl DotRpcApi for DotHttpRpcClient {
 		Ok(self.block(block_hash).await?.map(|block| block.block.extrinsics))
 	}
 
-	// TODO: When witnessing is catching up we query blocks in batches. It's posible that when
+	// TODO: When witnessing is catching up we query blocks in batches. It's possible that when
 	// a batch is made over a runtime boundary that the metadata will need to be queried more than
-	// necessary, as the order within the batch is not necessarily guaranteed. Beacause we limit
+	// necessary, as the order within the batch is not necessarily guaranteed. Because we limit
 	// Polkadot to 32 concurrent requests and runtime upgrades are infrequent this should not be an
 	// issue in reality, but probably worth solving at some point.
-	async fn events(&self, block_hash: PolkadotHash) -> Result<Option<Events<PolkadotConfig>>> {
-		let chain_runtime_version = self.runtime_version(Some(block_hash)).await?;
+	async fn events(
+		&self,
+		block_hash: PolkadotHash,
+		parent_hash: PolkadotHash,
+	) -> Result<Option<Events<PolkadotConfig>>> {
+		// We need to get the runtime version at the previous block instead the desired block
+		// because the events in the block are encoded using the previous block's runtime version,
+		// not the desired block's runtime version.
```

Comment:
we should mention that this is because the state_getRuntimeVersion RPC returns the runtime version at the end of the block